### PR TITLE
Do not insert non-recording spans in ETS

### PR DESCRIPTION
**Pull request by dvic**
As per the discussion in https://github.com/open-telemetry/opentelemetry-erlang/pull/263#issuecomment-896845262, this PR ensures that non-recording spans are not inserted anymore in ETS.

### CODE REVIEW

**Comment by tsloughter** [comment]
Thanks.

So yea, I still can't find a reason we'd need or want to store this span in ets.

Event `otel_span` functions ignore calls on spans that are not recording, so the table is never updated.

**Comment by bryannaegele** [comment]
Probably wouldn't hurt to add a non-recording span case in https://github.com/open-telemetry/opentelemetry-erlang/blob/main/apps/opentelemetry/test/opentelemetry_SUITE.erl#L149

File: apps/opentelemetry/src/otel_span_ets.erl [hunk 1]
```diff
@@ -55,10 +55,10 @@ start_span(Ctx, Name, Opts, Processors, InstrumentationLibrary) ->
             Span2 = Processors(Ctx, Span1),
             _ = storage_insert(Span2),
             SpanCtx;
-        {SpanCtx, Span=#span{}} ->
+        {SpanCtx, #span{}} ->
             %% span isn't recorded so don't run processors
             %% but we do insert to ets table?
-            _ = storage_insert(Span),
+            %% _ = storage_insert(Span),
```
Comment (tsloughter) [inline apps/opentelemetry/src/otel_span_ets.erl:61]:
```suggestion
```

File: apps/opentelemetry/src/otel_span_ets.erl [hunk 2]
```diff
@@ -55,10 +55,10 @@ start_span(Ctx, Name, Opts, Processors, InstrumentationLibrary) ->
             Span2 = Processors(Ctx, Span1),
             _ = storage_insert(Span2),
             SpanCtx;
-        {SpanCtx, Span=#span{}} ->
+        {SpanCtx, #span{}} ->
             %% span isn't recorded so don't run processors
             %% but we do insert to ets table?
```
Comment (tsloughter) [inline apps/opentelemetry/src/otel_span_ets.erl:60]:
```suggestion
            %% span isn't recorded so don't run processors or insert into ets table
```

**Review by tsloughter — CHANGES_REQUESTED** [review]

**Comment by dvic** [comment]
@tsloughter I applied your suggestions and rebased into one commit.

**Comment by dvic** [comment]
I've added a simple test case.

**Review by tsloughter — APPROVED** [review]

**Pull request merged**